pkg/storage/snapshot/snapshot_base.go

```diff
@@ -395,9 +396,13 @@ func (ctrl *VMSnapshotController) handleVMSnapshotContent(obj interface{}) {
 			log.Log.Errorf(failedKeyFromObjectFmt, err, content)
 			return
 		}
+		keys, err := ctrl.VMSnapshotInformer.GetIndexer().IndexKeys("vmSnapshotContent", objName)
+		if err != nil {
+			utilruntime.HandleError(err)
+			return
+		}
 
-		if content.Spec.VirtualMachineSnapshotName != nil {
-			k := cacheKeyFunc(content.Namespace, *content.Spec.VirtualMachineSnapshotName)
```

Comment:
Why is the index better?

Reply:
the index can be used when the content is already deleted -> nil in that case you can't get content.Spec.VirtualMachineSnapshotName